src/flashcart/sc64/sc64.c

```
@@ -126,13 +155,31 @@ static bool disk_system_lba_is_bad (uint16_t lba, flashcart_disk_parameters_t *d
+ * 
+ * @param offset Offset for the THB mapping.
+ * @param track Track number.
+ * @param head Head number.
+ * @param block Block number.
+ * @param valid Valid flag.
+ * @param writable Writable flag.
+ * @param file_offset File offset.
+ */
 static void disk_set_thb_mapping (uint32_t offset, uint16_t track, uint8_t head, uint8_t block, bool valid, bool writable, int file_offset) {
     uint32_t index = (track << 2) | (head << 1) | (block);
     uint32_t mapping = valid ? ((writable ? THB_WRITABLE_FLAG : 0) | (file_offset & ~(THB_WRITABLE_FLAG))) : THB_UNMAPPED;
```

Cppcheck (2.10-2)

[error] Signed integer overflow for expression '1<<31'.

(integerOverflow)